src/routes/posts.js

```diff
@@ -105,6 +105,66 @@ const fetchPosts = async (req, res, next) => {
     }
 }
 
+// ASSIGN TAG TO AN ARTICLE
+
+const assignTagToPost = async (req, res) => {
+    try {
+        const { postId } = req.params // post id;
+        const { tagId } = req.body
+
+        const postsTags = await db.query(
+            'INSERT INTO posts_tags ("postId","tagId") SELECT $1,$2 WHERE NOT EXISTS (SELECT * FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2) RETURNING *',
+            [postId, tagId]
+        )
+
+        if (!postsTags.rows[0]) {
+            res.status(400).json({
+                message: 'Tag is already assigned to the post',
+            })
+        } else {
+            res.status(201).json({
```

Comment:
the main resource for the route is a `post`. so `201` here would mean a new `post` has been created.

Although assigning a tag means an insert, just a `200` here is fine.

Reply:
resolved